### PR TITLE
chore: sonar-cloud ci

### PR DESCRIPTION
**Pull request by Laurin-W**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -13,13 +13,19 @@ jobs:
   ci:
     runs-on: ubuntu-latest
     steps:
+      - run: echo "node_version=`cat .nvmrc`" >> $GITHUB_ENV
       - uses: actions/checkout@v3
```
Comment (RWJMurphy) [inline .github/workflows/ci.yml:17]:
```suggestion
      - uses: actions/checkout@v3
      - run: echo "node_version=$(cat .nvmrc)" >> $GITHUB_ENV
```

Comment (Laurin-W) [reply .github/workflows/ci.yml]: Thanks for spotting! Fixed with https://github.com/inrupt/client-identifier-helper/pull/140/commits/834c88ea736d7776ef8c82c392d08ef7116af243

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -14,12 +14,18 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
+      - run: echo "node_version=$(cat .nvmrc)" >> $GITHUB_ENV
       - uses: actions/setup-node@v3
         with:
-          node-version-file: ".nvmrc"
+          node-version: ${{ env.node_version }}
```
Comment (ThisIsMissEm) [inline .github/workflows/ci.yml:20]: No, don't do this, use the `node-version-file`, it's really what you want here. You may have seen an set-output warning, you can ignore it, actions/setup-node just needs to be updated.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -14,12 +14,18 @@ jobs:
...
           cache: "npm"
       - run: npm ci --ignore-scripts
       - run: npm test
+      # Upload coverage for sonarcube (only matching OS and one node version required)
+      - uses: actions/upload-artifact@v3
+        with:
+          name: code-coverage-ubuntu-latest-${{ env.node_version }}
```
Comment (ThisIsMissEm) [inline .github/workflows/ci.yml:27]:
```suggestion
          name: code-coverage-report
```

The node version doesn't matter here.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -47,3 +53,23 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # Sonar analysis needs the full history for features like automatic assignment of bugs. If the following step
+          # is not included the project will show a warning about incomplete information.
+          fetch-depth: 0
+      - run: echo "node_version=$(cat .nvmrc)" >> $GITHUB_ENV
+      - uses: actions/download-artifact@v3
+        with:
+          name: code-coverage-ubuntu-latest-${{ env.node_version }}
```
Comment (ThisIsMissEm) [inline .github/workflows/ci.yml:70]:
```suggestion
          name: code-coverage-report
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -47,3 +53,23 @@ jobs:
...
+    if: ${{ github.actor != 'dependabot[bot]' }}
+    needs: [ci]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # Sonar analysis needs the full history for features like automatic assignment of bugs. If the following step
+          # is not included the project will show a warning about incomplete information.
+          fetch-depth: 0
+      - run: echo "node_version=$(cat .nvmrc)" >> $GITHUB_ENV
```
Comment (ThisIsMissEm) [inline .github/workflows/ci.yml:67]:
```suggestion
```

**Review by ThisIsMissEm — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Laurin-W) [reply .github/workflows/ci.yml]: changed with: https://github.com/inrupt/client-identifier-helper/pull/140/commits/ebd392ef2c0d289e4e4e1bb74bf6d6b83381024f

**Review by ThisIsMissEm — APPROVED** [review]

**Pull request merged**